auth/src/test/java/com/firebase/ui/auth/ui/idp/CredentialSignInHandlerTest.java

```diff
@@ -0,0 +1,191 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+
+import com.firebase.ui.auth.BuildConfig;
+import com.firebase.ui.auth.provider.IDPResponse;
+import com.firebase.ui.auth.test_helpers.ActivityHelperShadow;
+import com.firebase.ui.auth.test_helpers.AutoCompleteTask;
+import com.firebase.ui.auth.test_helpers.CustomRobolectricGradleTestRunner;
+import com.firebase.ui.auth.test_helpers.FakeAuthResult;
+import com.firebase.ui.auth.test_helpers.FakeProviderQueryResult;
+import com.firebase.ui.auth.test_helpers.TestConstants;
+import com.firebase.ui.auth.test_helpers.TestHelper;
+import com.firebase.ui.auth.ui.ActivityHelper;
+import com.firebase.ui.auth.ui.ExtraConstants;
+import com.firebase.ui.auth.ui.FlowParameters;
+import com.firebase.ui.auth.ui.account_link.SaveCredentialsActivity;
+import com.firebase.ui.auth.ui.account_link.WelcomeBackIDPPrompt;
+import com.firebase.ui.auth.ui.account_link.WelcomeBackPasswordPrompt;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.EmailAuthProvider;
+import com.google.firebase.auth.FacebookAuthProvider;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthUserCollisionException;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.ProviderQueryResult;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.annotation.Config;
+
+import java.util.Arrays;
+
+
+@RunWith(CustomRobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21, shadows = {ActivityHelperShadow.class})
+public class CredentialSignInHandlerTest {
+    private static final int RC_ACCOUNT_LINK = 3;
+    private static final int RC_SAVE_CREDENTIALS = 4;
+    private static final String LINKING_ERROR = "ERROR_TEST_LINKING";
+    private static final String LINKING_EXPLANATION = "Test explanation";
+
+    @Test
+    public void testSignInSucceeded() {
+        Activity mockActivty = mock(Activity.class);
+        ActivityHelper mockActivityHelper = mock(ActivityHelper.class);
+        FirebaseUser mockFirebaseUser = TestHelper.makeMockFirebaseUser();
+        IDPResponse mockIdpResponse = mock(IDPResponse.class);
+
+        CredentialSignInHandler credentialSignInHandler = new CredentialSignInHandler(
+                mockActivty,
+                mockActivityHelper,
+                RC_ACCOUNT_LINK,
+                RC_SAVE_CREDENTIALS,
+                mockIdpResponse);
+
+        Context mockContext = mock(Context.class);
+        FlowParameters mockFlowParams = mock(FlowParameters.class);
+        Task mockTask = mock(Task.class);
+
+        when(mockTask.isSuccessful()).thenReturn(true);
+        when(mockTask.getResult()).thenReturn(new FakeAuthResult(mockFirebaseUser));
```

Comment:
I think you should be able to use `Tasks.forResult(new FakeAuthResult(mockFirebaseUser));` instead of a mock object without consequence here, it should be more readable.